### PR TITLE
Patch/fix favorites isbn

### LINKED ISSUE #29: server.js- line 44- fix post bug

### LINKED ISSUE #30: change delete by isbn to delete by id

### PR DESCRIPTION
**Pull request by cagallo**
## What Features are Being Added?

- Added migration file to add a new column to book tables- isFavorited- with a default value of false
- Created updateFavorited() (queries.js) and PATCH network request for the query

## What is Being Refactored?

- cleaned console.logs out of migration files
- changed id to isbn for getSingleBook and removeBookFromFav

## Where should the reviewer start?

- server.js, queries.js, migrations/add_isFavorited_to_book_table.js

## How should this be manually tested?

- postman

## What are the relevant tickets?

- Add isFavorited to books table
- Create PATCH request to update isFavorited value

## Any Other Notes/Questions/Problems?

- None, let's get this deployed tomorrow!

### CODE REVIEW

File: migrations/20220224221948_add_isFavorited_to_book_table.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+exports.up = function(knex) {
+  return knex.schema.table('books', (table) => {
```
Comment (krdurrett) [inline migrations/20220224221948_add_isFavorited_to_book_table.js:2]: Added isFavorited, this is a great idea~

File: migrations/20220224221948_add_isFavorited_to_book_table.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    table.string('isFavorited').defaultTo(false)
```
Comment (Fordo29) [inline migrations/20220224221948_add_isFavorited_to_book_table.js:3]: Nice way to populate the data for isFavorited to start as false

File: queries.js [hunk 3]
```diff
@@ -17,5 +17,8 @@ module.exports = {
     },
     getAllFavs() {
       return database('favorites');
+    },
+    updateFavorited(request) {
+      return database('books').where('isbn', request.params.isbn).update({isFavorited: request.body.isFavorited});
```
Comment (krdurrett) [inline queries.js:22]: Nice query to alter the isFavorited property

Comment (Fordo29) [inline queries.js:22]: Extreme productive learning done here.. Thank you for driving us through this Chez

File: server.js [hunk 4]
```diff
@@ -62,12 +63,24 @@ app.delete('/api/v1/favorites/:isbn', (request, response) => {
...
         });
       }
     });
 });
 
+app.patch('/api/v1/books/:isbn', async (request, response) => {
+  queries.updateFavorited(request)
+    .then(count => {
+      if (count) {
+        response.status(200).json({message: `Book with isbn#${request.params.isbn} isFavorited: ${request.body.isFavorited}`})
```
Comment (krdurrett) [inline server.js:76]: Yes this patch looks great

File: server.js [hunk 5]
```diff
@@ -62,12 +63,24 @@ app.delete('/api/v1/favorites/:isbn', (request, response) => {
...
       } else {
         response.status(404).json({
-          error: `Could not find book with id ${request.params.isbn}`
+          error: `Could not find book with isbn ${request.params.isbn}`
         });
       }
     });
 });
 
+app.patch('/api/v1/books/:isbn', async (request, response) => {
```
Comment (Fordo29) [inline server.js:72]: Nice job on the patch

**Review by krdurrett — APPROVED** [review]
Great changes, thanks for cleaning up these files too!

**Pull request merged**